ska_sun/sun.py

```diff
@@ -708,3 +736,118 @@ def apply_sun_pitch_yaw(
     return Quat(q=qs)
+
+
+def get_att_for_sun_pitch_yaw(
+    pitch: float,
+    yaw: float,
+    time: CxoTimeLike = None,
+    sun_ra: float | None = None,
+    sun_dec: float | None = None,
+    coord_system: str = "spacecraft",
+    off_nom_roll: float = 0,
+) -> Quat:
+    """Get sun-pointed attitude for given sun pitch and yaw angles.
+
+    This function is the inverse of ``get_sun_pitch_yaw()``, where the attitude is
+    constrained to have the specified ``off_nom_roll`` angle (default=0).
+
+    Parameters
+    ----------
+    pitch : float
+        Sun pitch angle (deg)
+    yaw : float
+        Sun yaw angle (deg)
+    time : CxoTimeLike, optional
+        Time for sun position.
+    sun_ra : float, optional
+        Sun RA (deg) instead of time.
+    sun_dec : float, optional
+        Sun Dec (deg) instead of time.
+    coord_system : str, optional
+        Coordinate system for yaw ("spacecraft" | "ORviewer", default="spacecraft").
+    off_nom_roll : float, optional
+        Off-nominal roll angle (deg)
+
+    Returns
+    -------
+    att : Quat
+        Attitude quaternion.
+    """
+    if sun_ra is None or sun_dec is None:
+        sun_ra, sun_dec = position(time)
+
+    # Generate an attitude pointed at the north ecliptic pole. Since the sun is near the
+    # ecliptic plane this never has numerical problems and pitch0 is around 90 degress.
+    # Then apply the appropriate pitch and yaw offsets to get to the desired sun pitch
+    # and yaw.
+    roll = nominal_roll(0, 90, sun_ra=sun_ra, sun_dec=sun_dec) + off_nom_roll
+    att0 = Quat([0, 90, roll])
+    pitch0, yaw0 = get_sun_pitch_yaw(
+        att0.ra, att0.dec, sun_ra=sun_ra, sun_dec=sun_dec, coord_system=coord_system
+    )
+    att = apply_sun_pitch_yaw(
+        att0,
+        pitch=pitch - pitch0,
+        yaw=yaw - yaw0,
+        sun_ra=sun_ra,
+        sun_dec=sun_dec,
+        coord_system=coord_system,
+    )
+    return att
+
+
+def get_nsm_attitude(att: QuatLike, time: CxoTimeLike, pitch: float = 90) -> Quat:
+    """
+    Calculate the closest Normal Sun Mode attitude from starting attitude.
+
+    The default is for a NSM pitch of 90 degrees (pure minus-Z at sun). An arbitrary
+    offset pitch angle can be specified.
+
+    The calculation is based on the Chandra - sun vector in ECI. The function defines
+    the vector in the body frame that will be pointed at the sun. The normal sun mode
+    maneuver is then calculated as the shortest maneuver that points that vector at the
+    Sun. Note that for off-nominal roll, this is NOT a pure pitch maneuver.
+
+    Parameters
+    ----------
+    att : Quat
+        Attitude that can initialize a Quat().
+    time : CxoTimeLike
+        Time in a valid Chandra.Time format.
+    pitch : float, optional
+        NSM pitch angle in degrees. The default is 90.
+
+    Returns
+    -------
+    Quat
+        NSM attitude quaternion.
+    """
+    # Calc Chanrda - sun vector in ECI (ignore CXO orbit)
```

Comment:
Might as well fix the Chanrda typo while copying over.

Reply:
Done.